internal/cmd/workspaces.go

```diff
@@ -120,6 +131,201 @@ func lsWorkspacesCommand() *cobra.Command {
 	return cmd
 }
 
+func pingWorkspaceCommand() *cobra.Command {
+	var (
+		schemes []string
+		count   int
+	)
+
+	cmd := &cobra.Command{
+		Use:     "ping [workspace_name]",
+		Short:   "ping Coder workspaces by name",
+		Long:    "ping Coder workspaces by name",
+		Example: `coder workspaces ping front-end-workspace`,
+		Args:    xcobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return err
+			}
+			workspace, err := findWorkspace(ctx, client, args[0], coder.Me)
+			if err != nil {
+				return err
+			}
+
+			iceSchemes := map[ice.SchemeType]interface{}{}
+			for _, rawScheme := range schemes {
+				scheme := ice.NewSchemeType(rawScheme)
+				if scheme == ice.Unknown {
+					return fmt.Errorf("scheme type %q not recognized", rawScheme)
+				}
+				iceSchemes[scheme] = nil
+			}
+
+			pinger := &wsPinger{
+				client:     client,
+				workspace:  workspace,
+				iceSchemes: iceSchemes,
+			}
+
+			seq := 0
+			ticker := time.NewTicker(time.Second)
+			for {
+				select {
+				case <-ticker.C:
+					err := pinger.ping(ctx)
+					if err != nil {
+						return err
+					}
+					seq++
+					if count != 0 && seq >= count {
```

Comment:
```suggestion
					if count > 0 && seq >= count {
```

if count is 0 or negative, we don't want to exit... or we can validate the input parameter, ping seems to do that:

```shell-session
$ ping -c 0 google.com
ping: invalid argument: '0': out of range: 1 <= value <= 9223372036854775807
```